packages/block-library/src/table/state.js

```diff
@@ -78,11 +76,17 @@ export function updateSelectedCell( state, selection, updateCell ) {
 		return state;
 	}
 
-	const tableSections = pick( state, [ 'head', 'body', 'foot' ] );
+	const { head, body, foot } = state;
+	const tableSections = { head, body, foot };
 	const { sectionName: selectionSectionName, rowIndex: selectionRowIndex } =
 		selection;
 
 	return mapValues( tableSections, ( section, sectionName ) => {
+		// Bail early if the table section is empty.
+		if ( isEmptyTableSection( section ) ) {
+			return section;
+		}
```

Comment:
This is being done in all other places where we're handling the table sections, but not here. Seems like using `_.pick()` was concealing this, but now when we use destructuring and object composition we uncovered that issue. Adding the check here to avoid unexisting sections from handling.